packages/roosterjs-content-model-plugins/lib/imageEdit/utils/imageMetadata.ts

```diff
@@ -0,0 +1,67 @@
+import {
+    EditingInfoDatasetName,
+    ImageMetadataFormatDefinition,
+    validate,
+} from 'roosterjs-content-model-dom';
+import type { ImageMetadataFormat } from 'roosterjs-content-model-types';
+
+/**
+ * @internal
+ * Get metadata object from an HTML element
+ * @param element The HTML element to get metadata object from
+ * @param definition The type definition of this metadata used for validate this metadata object.
+ * If not specified, no validation will be performed and always return whatever we get from the element
+ * @param defaultValue The default value to return if the retrieved object cannot pass the validation,
+ * or there is no metadata object at all
+ * @returns The strong-type metadata object if it can be validated, or null
+ */
+export function getMetadata(element: HTMLElement): ImageMetadataFormat | null {
```

Comment:
We already have updateImageMetadata() to do the same job. So no need to add new API for this. Also no need to export EditingInfoDatasetName and ImageMetadataFormatDefinition